### ISSUE TITLE
Allow naming of JSObject method implementations.

### ISSUE THREAD

**Issue by adamjryan** [konsoletyper/teavm#303]
Currently any method implementation in a JSObject ends up with an indeterminately generated name in the javascript. It would be useful to specify method identifiers so that when they are transformed to javascript they can be called and used in external javascript code.
I'm not sure whether this already exists and just requires documentation, there appears to be an @Export annotation but this is only for web-assembly.

**Comment by konsoletyper** [comment]
No, currently there's no solution for this problem. There's one workaround: you can define your own subtype of `JSObject` like this:

```java
public interface MyExportedObject extends JSObject {
    void foo();
}
```

And implement it by a regular Java class:

```java
public class ExportedImpl implements MyExportedObject {
    @Override
    public void foo() {
        System.out.println("It works");
    }
}
```

Now, if you instantiate this class and pass the instance to a JavaScript function, it will be able to call `foo` method.

This approach does not work for static methods though.

Can you clarify your desired use case? What problem you are going to solve?

**Comment by adamjryan** [comment]
That does solve it to some degree. I have a couple of use cases, one is to pass an object defined in Java as the handler in a javascript Proxy. This requires certain methods to be defined that get called by the Proxy wrapper so it would be a bit easier to specify that the name should be used as is rather than needing to create an interface to use first.
The other case is where I want to create objects to be used as a library so it needs to have a defined interface that someone can call.
